pkg/planner/planner.go

```diff
@@ -25,4 +25,6 @@ type Procedure interface {
 	Do(ctx context.Context) ([]Procedure, error)
 	// ShouldDo determines if this procedure should be executed
 	ShouldDo(ctx context.Context) (bool, error)
+	// Verify the state after Do.
```

Comment:
nit: might be worth adding some more context as to why we might want to verify and that its only for certain situations.

Again, a nice to have but not essential.